### PR TITLE
Add feed block

### PR DESCRIPTION
**Pull request by sdmcraft**
This PR proposes a generic `feed` block for fetching feeds of any kind like etc.

Test URLs:
- Original: NA
- Before: https://main--sunstar--hlxsites.hlx.live
- After: https://i-9c--sunstar--hlxsites.hlx.live

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://<branch>--sunstar--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/<your-block>&index=0

### CODE REVIEW

**Comment by sdmcraft** [comment]
Seeking early feedback on a generic feed block to display feed of any kind.
For e.g. 

image

**Comment by bosschaert** [comment]
Since I think the idea is that this will be a generic piece it might be worth adding a unit test for the logic.

**Comment by sdmcraft** [comment]
> @sdmcraft Can you point me to the actual website page (if any) corresponding to the feed.

I am thinking of `feed` as a metablock which can be used to build a feed of any type of block by specifying the criteria and the block type. For e.g. on the home page, there's a `news feed` with `cards` block. Similarly we have `Related Articles` , `Featured Article` feed which could be based on the same. Basically all of them are feeds (dynamically derived from the index) and put inside a block (for e.g. cards block). 
The intent is to share this logic of querying up the index and rendering it in a user specified block in the UI.

File: blocks/feed/feed.js
Comment (jindaliiita) [inline blocks/feed/feed.js:35]: it would be good if support for custom logic here based on the feed type

File: blocks/feed/feed.js
Comment (jindaliiita) [inline blocks/feed/feed.js:10]: Also suppose we have to add custom fields to display for some specific fields lets suppose date, then i think that should also be derived on the basis of feed type.

File: blocks/feed/feed.js
Comment (sdmcraft) [reply blocks/feed/feed.js:35]: I think most of the logic can be achieved via formulating index queries. We can add customization support later as the need arises.

**Comment by sdmcraft** [comment]
> Since I think the idea is that this will be a generic piece it might be worth adding a unit test for the logic.

I wanted to add unit test for this but blocked on #72 . Would it be ok to take them up seperately?

File: blocks/feed/feed.js
Comment (sdmcraft) [reply blocks/feed/feed.js]: Good idea. Added.

File: test/blocks/feed/feed.test.js
Comment (sdmcraft) [inline test/blocks/feed/feed.test.js:1]: Test commented out till #72 is addressed.

**Comment by sdmcraft** [comment]
> > Since I think the idea is that this will be a generic piece it might be worth adding a unit test for the logic.
> 
> I wanted to add unit test for this but blocked on #72 . Would it be ok to take them up seperately?

@bosschaert , tests added now.

**Comment by sdmcraft** [comment]
> @sdmcraft Can you point me to the actual website page (if any) corresponding to the feed.

Here's an example added in the [block library](https://main--sunstar--hlxsites.hlx.page/sidekick/blocks/feed?view-doc-source=true). You can see the structure here, but to render it as expected, it has a dependency on indexing the required items (category, topic etc.) . I think we need to take that up separately.

File: head.html
Comment (bosschaert) [inline head.html:19]: Why the indirection via load.js which in turn calls scripts.js?

File: blocks/feed/feed.js
Comment (bosschaert) [inline blocks/feed/feed.js:6]: What is the resultParsers for? Maybe add a comment to explain a little...

File: head.html
Comment (sdmcraft) [reply head.html:19]: > Why the indirection via load.js which in turn calls scripts.js?

I had to do this for unit tests. So `feed.js` imports `scripts.js`. As a result, running a UT for `feed.js` implicitly loads `scripts.js` . Now `scripts.js` has `loadPage` getting called as soon as its loaded and this causes the whole front end rendering pipeline  to kick which requires too many mocks et al to run as a UT. And anyway the intent of the test is to just test the `feed` block and not the complete rendering pipeline. 
So I had 2 options:
- Move dependencies from `scripts.js` to some `utils.js` and import that in `feed.js`
- OR, move the `loadPage` to another file so that loading `scripts.js` doesn't end up kicking of the frontend rendering pieline.

The former was too disruptive, requiring lot of refactoring everywhere so I opted for latter. HTH.

File: blocks/feed/feed.js
Comment (sdmcraft) [reply blocks/feed/feed.js:8]: > What is the resultParsers for? Maybe add a comment to explain a little...

Good idea. Done.

**Comment by sdmcraft** [comment]
@jindaliiita , @bosschaert , for a visual reference, please see [this page](https://news-cards--sunstar--hlxsites.hlx.live/_drafts/satyam/). This page has a news cards feed formed by the feed block.

Please let me know if this can be merged of if there anything you'd want to addressed before that.

File: head.html
Comment (jindaliiita) [reply head.html:19]: @sdmcraft You can check [tags.test](https://github.com/hlxsites/sunstar/blob/main/test/blocks/tags/tags.test.js). And i think its not good that for tests only we are changing the default boilerplate behaviour.

File: blocks/feed/feed.js
Comment (jindaliiita) [inline blocks/feed/feed.js:55]: Nit: can use ternary operator here

File: blocks/feed/feed.js
Comment (jindaliiita) [inline blocks/feed/feed.js:53]: you can optimize this 
[Nit]: return (!category || category === el.category) && (!topic || topic === el.topic) && (!modifier || el.modifier === modifier)

**Review by jindaliiita — APPROVED** [review]
LGTM other than changes corresponding to the **type** and **category**.

**Pull request merged**
